tools/msconvert/msconvert_macros.xml

```diff
@@ -21,40 +22,80 @@
       ##set basename = $re.sub(r'[\/\\\;\|\&\>\<]','_',$input.display_name)
     #end if
 
-    msconvert_wrapper.py
-
     #if $ext == 'wiff':
-      --input='${input.extra_files_path}/wiff'
-      --input_name='${basename}.wiff'
-      --implicit='${input.extra_files_path}/wiff_scan'
-      --input='${input.extra_files_path}/wiff_scan'
-      --input_name='${basename}.wiff.scan'
+      ln -s '${input.extra_files_path}/wiff' '${basename}.wiff' &&
+      ln -s '${input.extra_files_path}/wiff_scan' '${basename}.wiff.scan' &&
+    #elif $ext.endswith('tar'):
+      ln -s '$input' '${basename}' &&
+      tar xf '${basename}' &&
+      #set basename = $os.path.splitext($basename)[0]
+    #else
+      ln -s '$input' '${basename}' &&
+    #end if
+    
+    #if $data_processing.precursor_refinement.use_mzrefinement
+      #set input_ident_name = ".".join((os.path.splitext($basename)[0], $data_processing.precursor_refinement.input_ident.ext))
+      #set output_refinement_name = os.path.splitext($basename)[0] + '.mzRefinement.tsv'
+      ln -s '$data_processing.precursor_refinement.input_ident' '$input_ident_name' &&
+    #end if
+
+    wine64 msconvert '${basename}'
+    --outdir `dirname ${output}`
```

Comment:
This is specifying the galaxy database dir. This is dangerous as msconvert could write into this folder without Galaxy knowing. So these files will live there forever.

I would recommend using `./out/` here and move the file to the final destination. For --outfile I would use `./out/foo.dat` or something similar.

Reply:
You're worried that msconvert will write some file there other than the named one (--outfile)? It won't but I admit it requires knowledge of its internals to not worry about that. A bigger issue is I didn't yet add support for the `multi_run_output` because I'm not sure what that would look like. It would certainly change the outdir/outfile (outfile would cause all the files to be written to the same name). In that case all the files for the collection would need to be dynamically collected from outdir, so it wouldn't make sense to write directly to `dirname ${output}`. Can you help me with what the `<outputs>` element should look like in the `multi_run_output==true` case?

Reply:
Yes I'm worried. Anything against using the current-working-dir?

What about https://github.com/galaxyproject/galaxy/blob/dev/test/functional/tools/collection_split_on_column.xml#L9 ?